src/ddprof_module_lib.cc

```diff
@@ -15,12 +15,33 @@
 #include <fcntl.h>
 #include <gelf.h>
 #include <libelf.h>
-#include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 namespace ddprof {
 
+class FileDescriptorHolder {
+public:
+  FileDescriptorHolder() : _fd(-1) {}
+  DDRes open_file(std::string_view path) {
```

Comment:
I think it would be best to follow `unique_ptr` API and having the resource creation outside the class